packages/crayons-core/src/utils/draggable.ts

```diff
@@ -0,0 +1,302 @@
+      return;
+    }
+    const sortContainer = e.composedPath().find((el) => el.id === this.host.id);
+    if (sortContainer && sortContainer !== this.previousContainer) {
+      // the drag element have entered or re-entered current container(this.host)
+      this.cancelDebouncedDrag = false;
+    }
+    this.previousContainer = sortContainer;
+  };
+
+  private onDragLeave = (e) => {
+    if (!dragElement) {
+      return;
+    }
+    const outTarget = e.fromElement || e.relatedTarget;
+    if (!e.currentTarget.contains(outTarget)) {
+      // Check whether the outTarget's host (in case of shadow dom) exists in currentTarget
+      const parentHost = this.getMatchingHost(
+        outTarget,
+        this.host.children[0].tagName
+      );
+      if (!e.currentTarget.contains(parentHost)) {
+        // the drag element have left the current container(this.host)
+        this.previousContainer = undefined;
+        this.cancelDebouncedDrag = true;
+
+        if (dragElement.parentElement.id !== this.host.id) {
+          // dragElement from another container
+          this.removePlaceholder();
+        } else {
+          // Sort within the same container
+          this.addElement(dragElement, this.nextSibling);
+        }
+      }
+    }
+  };
+
+  private onDragOver = (e) => {
+    e.preventDefault();
+    if (!dragElement) {
+      return;
+    }
+    const sortContainerId = dragElement.parentElement.id;
+    if (this.acceptFrom.includes(sortContainerId)) {
+      this.debouncedSetElement(this.childElements, dragElement, e.clientY);
+    }
+  };
+
+  // Both dragend and drop need to used as the drop will be fired only on the container on which the drag is dropped
+  // and no on the container where drag is originated.
+  private onDragEnd = (e) => {
+    this.resetData(e);
+  };
+
+  private onDrop = (e) => {
+    if (!dragElement) {
+      return;
+    }
+    const sortContainerId = dragElement.parentElement.id;
+    if (!this.acceptFrom.includes(sortContainerId)) {
+      return;
+    }
+    const newElement = this.placeholder || dragElement;
+    const droppedIndex = [...this.host.children].indexOf(newElement);
+    if (this.placeholder) {
+      if (this.options.addOnDrop) {
+        const clone = cloneNodeWithEvents(dragElement, true, true);
+        this.placeholder.replaceWith(clone);
+      } else {
+        this.removePlaceholder();
+      }
+    }
+    this.host.dispatchEvent(
+      new CustomEvent('fwDropBase', {
+        cancelable: true,
+        bubbles: false,
+        detail: {
+          droppedElement: dragElement,
+          droppedIndex,
+          dragFromId: sortContainerId,
+          dropToId: this.host.id,
+        },
+      })
+    );
+    this.resetData(e);
+  };
+
+  addListeners() {
+    this.host.addEventListener('dragstart', (e) => this.onDragStart(e));
+    this.host.addEventListener('dragend', (e) => this.onDragEnd(e));
+    this.host.addEventListener('dragenter', (e) => this.onDragEnter(e));
```

Comment:
you haven't removed the dragenter and dragleave listeners!

Reply:
Didn't notice this. Thanks for the catch:)